src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -654,6 +660,54 @@ - (void)setPairingDelegate:(id<MTRDevicePairingDelegate>)delegate queue:(dispatc
     });
 }
 
+- (void)setNocChainIssuer:(id<NOCChainIssuer>)nocChainIssuer
+{
+    dispatch_sync(_chipWorkQueue, ^{
+        self->_operationalCredentialsDelegate->SetNocChainIssuer(nocChainIssuer);
+    });
+}
+
+- (uint32_t)onNOCChainGeneration:(NSData *)operationalCertificate
+         intermediateCertificate:(NSData *)intermediateCertificate
+                 rootCertificate:(NSData *)rootCertificate
+                             ipk:(NSData *)ipk
+                    adminSubject:(uint64_t)adminSubject
+{
+    // use ipk and adminSubject from CommissioningParameters if not passed in
+    chip::Controller::CommissioningParameters commissioningParams = _autoCommissioner->GetCommissioningParameters();
+
+    chip::Optional<chip::Crypto::AesCcm128KeySpan> ipkOptional;
+    uint8_t ipkValue[chip::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
+    chip::Crypto::AesCcm128KeySpan ipkTempSpan(ipkValue);
+    chip::ByteSpan ipkBytes = AsByteSpan(ipk);
+    if (ipk != nil) {
+        VerifyOrReturnValue(ipkBytes.size() == sizeof(ipkValue), CHIP_ERROR_INTERNAL.AsInteger());
+        memcpy(&ipkValue[0], ipkBytes.data(), ipkBytes.size());
+        ipkOptional.SetValue(ipkTempSpan);
+    } else if (commissioningParams.GetIpk().HasValue()) {
+        ipkOptional.SetValue(commissioningParams.GetIpk().Value());
+    }
+
+    chip::Optional<chip::NodeId> adminSubjectOptional;
+    if (adminSubject == chip::kUndefinedNodeId) {
```

Comment:
Should be using `NSNumber` and passing nil as needed.

Reply:
Done